Python/pystrtod.c

```diff
@@ -918,6 +932,97 @@ char * PyOS_double_to_string(double val,
 
 /* _Py_dg_dtoa is available. */
 
+/* turn ASCII hex characters into integer values and vice versa */
+
+static char
+char_from_hex(int x, int upper)
+{
+    assert(0 <= x && x < 16);
+    char c = Py_hexdigits[x];
+    if (upper) {
+        c = Py_TOUPPER(c);
```

Comment:
That looks slow, it would be better to have an uppercase flavor of Py_hexdigits, such as Py_hexdigits_upper.

Reply:
In fact, this helper was literally moved from the ``floatobject.c``.

Reply:
@vstinner, I've added ``Py_hexdigits_upper``.  But I'm not sure if it worth (maybe as a local constant?): e.g. there are no other use cases across the codebase.